pkg/cloud/libvirt/client/client.go

```diff
@@ -120,21 +114,33 @@ type Client interface {
 
 type libvirtClient struct {
 	connection *libvirt.Connect
+
+	// storage pool that holds all volumes
+	pool *libvirt.StoragePool
+	// cache pool's name so we don't have to call failable GetName() method on pool all the time.
+	poolName string
```

Comment:
@bison Hi, any opinion on this change: making the storage pool part of the client so we don't need to fetch it all the time and pass the pool or name to all the volume handling functions everywhere? I first thought it'd be a good idea and hence the change here but now I'm not so sure.

Reply:
> I first thought it'd be a good idea and hence the change here but now I'm not so sure

Could you expand on that? 

Otherwise this /lgtm but just wanted to understand why you mentioned that you were not so sure now.

Reply:
I wasn't sure if there is a 1-1 relationship between `libvirtClient` and storage pools in terms of lifecycle. I asked @ingvagabund on slack yesterday and they said it's ok.